metagraph/src/annotation/taxonomy/tax_classifier.cpp

```diff
@@ -0,0 +1,300 @@
+                      filepath);
+        exit(1);
+    }
+
+    tsl::hopscotch_set<std::string> input_accessions;
+    if (anno_matrix != NULL) {
+        for (const std::string &label : anno_matrix->get_annotation().get_all_labels()) {
+            input_accessions.insert(get_accession_version_from_label(label));
+        }
+    }
+
+    while (getline(f, line)) {
+        if (line == "") {
+            logger->error("Error: The accession to taxid map table contains empty lines. "
+                          "Please make sure that this file was not manually modified {}", filepath);
+            exit(1);
+        }
+        std::vector<std::string> parts = utils::split_string(line, "\t");
+        if (parts.size() <= 2) {
+            logger->error("Error: The accession to taxid map table contains incomplete lines. "
+                          "Please make sure that this file was not manually modified {}", filepath);
+            exit(1);
+        }
+        if (input_accessions.size() == 0 || input_accessions.count(parts[1])) {
+            accversion_to_taxid_map_[parts[1]] = std::stoul(parts[2]);
+        }
+    }
+}
+
+TaxonomyClsAnno::TaxonomyClsAnno(const graph::AnnotatedDBG &anno,
+                                 const double lca_coverage_rate,
+                                 const double kmers_discovery_rate,
+                                 const std::string &tax_tree_filepath,
+                                 const std::string &label_taxid_map_filepath)
+             : TaxonomyBase(lca_coverage_rate, kmers_discovery_rate),
+               anno_matrix_(&anno) {
+    if (!std::filesystem::exists(tax_tree_filepath)) {
+        logger->error("Error: Can't open taxonomic tree file {}", tax_tree_filepath);
+        exit(1);
+    }
+
+    // Take one sample label and find the label type.
+    std::string sample_label = anno_matrix_->get_annotation().get_all_labels()[0];
+
+    // If true, require_accversion_to_taxid_map means that the taxid is not mentioned as part of the label. Thus, the program
+    // needs to parse an additional accession_version to taxid lookup file (optional argument: label_taxid_map_filepath).
+    bool require_accversion_to_taxid_map;
+    if (utils::starts_with(sample_label, "gi|")) {
+        // e.g.   >gi|1070643132|ref|NC_031224.1| Arthrobacter phage Mudcat, complete genome
+        label_type_ = GEN_BANK;
+        require_accversion_to_taxid_map = true;
+    } else if (utils::starts_with(utils::split_string(sample_label, ":")[1], "taxid|")) {
+        // e.g.   >kraken:taxid|2016032|NC_047834.1 Alteromonas virus vB_AspP-H4/4, complete genome
+        label_type_ = TAXID;
+        require_accversion_to_taxid_map = false;
+    } else {
+        logger->error("Error: Can't determine the type of the given label {}. "
+                      "Make sure the labels are in a recognized format.", sample_label);
+        exit(1);
+    }
+
+    Timer timer;
+    if (require_accversion_to_taxid_map) {
```

Comment:
Some code is redundant
```suggestion
    if (utils::starts_with(sample_label, "gi|")) {
        // e.g.   >gi|1070643132|ref|NC_031224.1| Arthrobacter phage Mudcat, complete genome
        label_type_ = GEN_BANK;
    } else if (utils::starts_with(utils::split_string(sample_label, ":")[1], "taxid|")) {
        // e.g.   >kraken:taxid|2016032|NC_047834.1 Alteromonas virus vB_AspP-H4/4, complete genome
        label_type_ = TAXID;
    } else {
        logger->error("Error: Can't determine the type of the given label {}. "
                      "Make sure the labels are in a recognized format.", sample_label);
        exit(1);
    }

    Timer timer;
    if (label_type_ == GEN_BANK) {
```